### PR TITLE
Clean up documentation

### PR DESCRIPTION
**Pull request by technovangelist**
Will probably need to update with PRs for new release.

This accomplishes a few things. 

- First it looks at the api docs, makes them a bit more consistent, and fixes the requests and responses to reflect how they actually work today. 
- It creates a better README for the docs folder
- Creates a DeepDive for understanding the files and layers using the api (inspired by the gdev doc Jeff shared)
- Creates a troubleshooting guide that starts to share the common solutions to some error messages that pop up.
- Moves some of the FAQ questions into more appropriate places in the docs.

I think that’s it.

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# How to troubleshoot issues
+
+Sometimes Ollama may not perform as you expect. One of the best ways to figure out what just happened is to take a look at the logs. You can find the logs on Mac by running the command:
+
+```shell
+cat ~/.ollama/logs/server.log
```
Comment (rgaidot) [inline docs/troubleshooting.md:6]: or `tail -f`?

**Comment by rgaidot** [comment]
Why not use absolute links instead of relative links throughout the entire documentation?

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+```
+
+On Linux, the logs can be found with this command:
```
Comment (rgaidot) [inline docs/troubleshooting.md:9]: if we use systemctl

File: README.md [hunk 3]
```diff
@@ -125,28 +125,52 @@ For more examples, see the [examples](examples) directory. For more information
...
+
 ```
 ollama pull llama2
 ```
 
-> This command can also be used to update a local model. Only the diff will be pulled.
-
 ### Remove a model
 
+Remove the model files from your local system.
```
Comment (jmorganca) [inline README.md:144]: I think we can do without this one

File: README.md [hunk 4]
```diff
@@ -125,28 +125,52 @@ For more examples, see the [examples](examples) directory. For more information
...
+
 ```
 ollama rm llama2
 ```
 
 ### Copy a model
 
+Copy a model on your local system, giving it a new name. The underlying files, such as the model weights, are not duplicated.
```
Comment (jmorganca) [inline README.md:152]: Similar here, I think it's mostly self-explanatory

File: README.md [hunk 5]
```diff
@@ -125,28 +125,52 @@ For more examples, see the [examples](examples) directory. For more information
...
+
+Show the information about a model. You must include an option for what you want to show: template, modelfile, system, parameters, or license.
+
+```
+ollama show --modelfile llama2
+```
+
+### List local models
+
+List all the models you have pulled.
```
Comment (jmorganca) [inline README.md:168]: This one as well!

File: README.md [hunk 6]
```diff
@@ -165,15 +189,17 @@ $ ollama run llama2 "Summarize this file: $(cat README.md)"
  Ollama is a lightweight, extensible framework for building and running language models on the local machine. It provides a simple API for creating, running, and managing models, as well as a library of pre-built models that can be easily used in a variety of applications.
 ```
 
-### List models on your computer
+### Start Ollama
```
Comment (jmorganca) [inline README.md:192]:
```suggestion
### Manually start Ollama
```

File: README.md [hunk 7]
```diff
@@ -165,15 +189,17 @@ $ ollama run llama2 "Summarize this file: $(cat README.md)"
...
+
+If you installed Ollama using the installations on the website, you will usually not need to do this. If you built Ollama yourself, or you need to change some parameters, it may be necessary to run the service manually:
```
Comment (jmorganca) [inline README.md:194]: This is definitely a bit confusing today. I think a more descriptive title might help. This is a great place to introduce `OLLAMA_HOST` as well (vs this paragraph)

File: docs/troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+```shell
+journalctl -u ollama
+```
+
+If you need help interpreting what you see, try signing in to the Discord at [discord.gg/ollama](https://discord.gg/ollama).
+
+## Error Index
```
Comment (jmorganca) [inline docs/troubleshooting.md:17]:
```suggestion
## Known issues
```

**Review by jmorganca — COMMENTED** [review]
some initial comments

File: README.md (hunk 3 above)
Comment (technovangelist) [reply README.md:144]: Remove model all together or just the explanation?

File: docs/troubleshooting.md (hunk 8 above)
Comment (technovangelist) [reply docs/troubleshooting.md:17]: Known issues to me means problems with the product we know about but need to fix, but these should be all the error messages that come up and what they mean.

File: README.md (hunk 3 above)
Comment (technovangelist) [reply README.md:144]: Actually I think all the explanations of cli commands can be left out.

File: README.md (hunk 7 above)
Comment (technovangelist) [reply README.md:194]: I was also thinking it would be good to have a doc called something like hosting... It can go over all the questions in the faq and show examples of ollama_host etc

File: docs/troubleshooting.md (hunk 2 above)
Comment (technovangelist) [reply docs/troubleshooting.md:9]: added something about this.

File: README.md (hunk 4 above)
Comment (technovangelist) [reply README.md:152]: we have gotten a few questions in the discord about whether a copy 'copies' the files which it doesn't.

File: docs/troubleshooting.md (hunk 1 above)
Comment (technovangelist) [reply docs/troubleshooting.md:6]: Often the error is spread across a number of lines and having to run this a few times expanding the range, and explain the cli args to do so, is harder than just doing a single cat. But thanks for the comment

File: README.md [hunk 9]
```diff
@@ -59,7 +59,7 @@ Here are some example open-source models that can be downloaded:
 | Vicuna             | 7B         | 3.8GB | `ollama run vicuna`            |
 | LLaVA              | 7B         | 4.5GB | `ollama run llava`             |
 
-> Note: You should have at least 8 GB of RAM to run the 3B models, 16 GB to run the 7B models, and 32 GB to run the 13B models.
+> Note: You should have at least 8 GB of RAM to run the 7B models, 16 GB to run the 13B models, and 32 GB to run the 33B models.
```
Comment (jmorganca) [inline README.md:62]: I think the numbers are right, but we should clarify that you should have _available_ RAM

```suggestion
> Note: You should have at least 8 GB of RAM available to run the 7B models, 16 GB to run the 13B models, and 32 GB to run the 33B models.
```

File: README.md [hunk 10]
```diff
@@ -126,16 +126,20 @@ For more examples, see the [examples](examples) directory. For more information
 
 ### Create a model
 
-`ollama create` is used to create a model from a Modelfile.
+Create a model from a Modelfile
```
Comment (jmorganca) [inline README.md:129]: We can remove this section for the `Customize a Prompt` section below

File: README.md [hunk 11]
```diff
@@ -144,10 +148,26 @@ ollama rm llama2
...
 
+Copy a model on your local system, giving it a new name. The underlying files, such as the model weights, are not duplicated.
+
 ```
 ollama cp llama2 my-llama2
 ```
 
+### Show model information
+
+Show the information about a model. You must include an option for what you want to show: template, modelfile, system, parameters, or license.
```
Comment (jmorganca) [inline README.md:159]:
```suggestion
Show the information about a model. You must include an option for what you want to show: `template`, `modelfile`, `system`, `parameters`, or `license`.
```

File: README.md [hunk 12]
```diff
@@ -173,15 +193,19 @@ $ ollama run llama2 "Summarize this file: $(cat README.md)"
...
+
+By default, the Ollama service starts automatically. If you choose to build Ollama, you may need to manually run the service:
 
 ```
-ollama list
+ollama serve
 ```
 
-### Start Ollama
+There are a few environment variables you can use with `ollama serve` to influence how it functions, including `OLLAMA_HOST`, `OLLAMA_MODELS`, and more.
```
Comment (jmorganca) [inline README.md:204]: I would switch this to a link to the FAQ on customizing `ollama serve`

File: README.md [hunk 13]
```diff
@@ -173,15 +193,19 @@ $ ollama run llama2 "Summarize this file: $(cat README.md)"
...
 
 ```
-ollama list
+ollama serve
 ```
 
-### Start Ollama
+There are a few environment variables you can use with `ollama serve` to influence how it functions, including `OLLAMA_HOST`, `OLLAMA_MODELS`, and more.  
+
+## Upgrading Ollama
```
Comment (jmorganca) [inline README.md:206]: Move me to FAQ

File: docs/deepdive.md [hunk 14]
```diff
@@ -0,0 +1,126 @@
+# Ollama Deep Dive
```
Comment (jmorganca) [inline docs/deepdive.md:1]: I like some of the content here but after reviewing, most of it belongs in the FAQ to help specific questions (vs a broader deep dive topic)

File: docs/faq.md [hunk 15]
```diff
@@ -1,154 +1,9 @@
 # FAQ
```
Comment (jmorganca) [inline docs/faq.md:2]: The FAQ is still the right spot for common questions.

File: docs/server.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
+# Serving Ollama
```
Comment (jmorganca) [inline docs/server.md:1]: I need to think about this one a bit – i.e. this might be better off being the CLI documentation - of which `ollama serve` is an important comand

File: docs/faq.md (hunk 15 above)
Comment (technovangelist) [reply docs/faq.md:2]: We had discussed in standup removing most questions from faq and moving them to more appropriate places in docs

File: README.md (hunk 10 above)
Comment (technovangelist) [reply README.md:129]: removed in upcoming commit

File: README.md [hunk 17]
```diff
@@ -120,68 +120,11 @@ ollama run mario
 Hello! It's your friend Mario.
 ```
 
-For more examples, see the [examples](examples) directory. For more information on working with a Modelfile, see the [Modelfile](docs/modelfile.md) documentation.
+For more examples, see the [examples](examples) directory. For more information on working with a Modelfile, see the [Modelfile](docs/modelfile.md) documentation. For details about using the CLI, refer to our [CLI Documentation](./docs/cli.md).
 
-## CLI Reference
+## Upgrading Ollama
 
-### Create a model
```
Comment (jmorganca) [inline README.md:127]: Why was this deleted? We should still show the quick example commands here in the frontpage readme

File: docs/troubleshooting.md [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+
+```shell
+journalctl -u ollama
+```
+
+If you are manually running `ollama serve` in a terminal, the logs will be on that terminal.
+
+If you need help interpreting what you see, try signing in to the Discord at [discord.gg/ollama](https://discord.gg/ollama).
+
+## Known issues
```
Comment (jmorganca) [inline docs/troubleshooting.md:19]: This is a good spot to link to major known issues (e.g. ones we've closed repeated issues around)

File: docs/faq.md [hunk 19]
```diff
@@ -1,153 +1,43 @@
...
-
-### How can I change where Ollama stores models?
-
-To modify where models are stored, you can use the `OLLAMA_MODELS` environment variable. Note that on Linux this means defining `OLLAMA_MODELS` in a drop-in `/etc/systemd/system/ollama.service.d` service file, reloading systemd, and restarting the ollama service.
+See [the CLI Documentation](./cli.md) for more on this.
 
 ## Does Ollama send my prompts and answers back to Ollama.ai to use in any way?
 
-No. Anything you do with Ollama, such as generate a response from the model, stays with you. We don't collect any data about how you use the model. You are always in control of your own data.
+No, Ollama runs 100% locally wherever you have installed it. Other than downloading the models, we never reach out anywhere, and models are never updated using your questions and answers.
```
Comment (jmorganca) [inline docs/faq.md:28]:
```suggestion
No, Ollama runs entirely locally, and conversation data will never leave your machine.
```

File: docs/faq.md [hunk 20]
```diff
@@ -1,153 +1,43 @@
...
-systemctl restart ollama
-```
+You will need to use `OLLAMA_ORIGINS` to allow additional web origins. See [Serving Ollama in the CLI Documentation](./cli.md#serving-ollama) for more on using this.
 
 ## Where are models stored?
 
-- macOS: Raw model data is stored under `~/.ollama/models`.
-- Linux: Raw model data is stored under `/usr/share/ollama/.ollama/models`
+- macOS: All model data is stored under `~/.ollama/models`.
+- Linux: All model data is stored under `/usr/share/ollama/.ollama/models`
```
Comment (jmorganca) [inline docs/faq.md:22]:
```suggestion
- macOS: `~/.ollama/models`.
- Linux: `/usr/share/ollama/.ollama/models`
```

File: docs/cli.md [hunk 21]
```diff
@@ -0,0 +1,206 @@
+# Ollama CLI
+
+There are two main functions of the Ollama CLI. First it runs the service on the backend. This is the process that actually interacts with the models. Then it's the main UI for the user, passing commands to the service to run.
```
Comment (jmorganca) [inline docs/cli.md:3]: For this iteration let's keep it a light reference

```suggestion
```

File: docs/api.md [hunk 22]
```diff
@@ -611,13 +747,13 @@ curl http://localhost:11434/api/show -d '{
...
-  "template": "\nUSER:{{ .Prompt }}\nASSISTANT:\n",
-  "license:": "<license>",
+  "modelfile": "# Modelfile generated by \"ollama show\"\n# To build a new Modelfile based on this one, replace the FROM line with:\n# FROM llava:latest\n\nFROM /Users/matt/.ollama/models/blobs/sha256:200765e1283640ffbd013184bf496e261032fa75b99498a9613be4e94d63ad52\nTEMPLATE \"\"\"{{ .System }}\nUSER: {{ .Prompt }}\nASSSISTANT: \"\"\"\nPARAMETER num_ctx 4096\nPARAMETER stop \"\u003c/s\u003e\"\nPARAMETER stop \"USER:\"\nPARAMETER stop \"ASSSISTANT:\"",
+  "parameters": "num_ctx                        4096\nstop                           \u003c/s\u003e\nstop                           USER:\nstop                           ASSSISTANT:",
+  "template": "{{ .System }}\nUSER: {{ .Prompt }}\nASSSISTANT: ",
   "details": {
     "format": "gguf",
-    "families": ["llama", "clip"],
+    "family": "llama",
+    "families": [ "llama", "clip" ],
```
Comment (jmorganca) [inline docs/api.md:756]:
```suggestion
    "families": ["llama", "clip"],
```

File: docs/api.md [hunk 23]
```diff
@@ -481,7 +588,11 @@ Create a model from a [`Modelfile`](./modelfile.md). It is recommended to set `m
 
 ### Examples
 
-#### Request
+#### Create a new model
+
+Create a new model from a modelfile.
```
Comment (jmorganca) [inline docs/api.md:593]:
```suggestion
Create a new model from a `Modelfile`.
```

File: docs/api.md [hunk 24]
```diff
@@ -464,13 +550,34 @@ curl http://localhost:11434/api/chat -d '{
...
+```
+
 ## Create a Model
 
 ```shell
 POST /api/create
 ```
 
-Create a model from a [`Modelfile`](./modelfile.md). It is recommended to set `modelfile` to the content of the Modelfile rather than just set `path`. This is a requirement for remote create. Remote model creation should also create any file blobs, fields such as `FROM` and `ADAPTER`, explicitly with the server using [Create a Blob](#create-a-blob) and the value to the path indicated in the response.
+Create a model from a [`Modelfile`](./modelfile.md). It is recommended to set `modelfile` to the content of the Modelfile rather than just set `path`. This is a requirement for remote create. Remote model creation must also create any file blobs, fields such as `FROM` and `ADAPTER`, explicitly with the server using [Create a Blob](#create-a-blob) and the value to the path indicated in the response. Note that this just creates the model on the Ollama server and doesn't push it to Ollama.ai.
```
Comment (jmorganca) [inline docs/api.md:580]:
```suggestion
Create a model from a [`Modelfile`](./modelfile.md). It is recommended to set `modelfile` to the content of the Modelfile rather than just set `path`. This is a requirement for remote create. Remote model creation must also create any file blobs, fields such as `FROM` and `ADAPTER`, explicitly with the server using [Create a Blob](#create-a-blob) and the value to the path indicated in the response.
```

File: docs/api.md [hunk 25]
```diff
@@ -425,7 +508,8 @@ A stream of JSON objects is returned:
   "created_at": "2023-08-04T08:52:19.385406455-07:00",
   "message": {
     "role": "assisant",
-    "content": "The"
+    "content": "The", 
+    "images": null
```
Comment (jmorganca) [inline docs/api.md:512]:
```suggestion
    "content": "The"
```

File: docs/api.md [hunk 26]
```diff
@@ -382,18 +426,57 @@ Final response:
...
+  "prompt_eval_count": 26,
+  "prompt_eval_duration": 383809000,
+  "eval_count": 298,
+  "eval_duration": 4799921000
+}
+```
+
+#### Chat request (With History)
+
+Send a chat message with a conversation history. You can use this same approach to start the conversation using multi-shot prompting.
```
Comment (jmorganca) [inline docs/api.md:477]:
```suggestion
Send a chat message with a conversation history. You can use this same approach to start the conversation using multi-shot or chain-of-thought prompting.
```

File: docs/api.md [hunk 27]
```diff
@@ -343,7 +384,9 @@ Advanced parameters (optional):
 
 ### Examples
 
-#### Request
+#### Chat Request (Streaming)
```
Comment (jmorganca) [inline docs/api.md:387]: I think we might have left out the contents of this section

File: docs/api.md [hunk 28]
```diff
@@ -321,7 +362,7 @@ curl http://localhost:11434/api/generate -d '{
 POST /api/chat
 ```
 
-Generate the next message in a chat with a provided model. This is a streaming endpoint, so there will be a series of responses. The final response object will include statistics and additional data from the request.
+Generate the next message in a chat with a provided model. This is a streaming endpoint, so there will be a series of responses. Streaming can be disabled using `stream: false`. The final response object will include statistics and additional data from the request.
```
Comment (jmorganca) [inline docs/api.md:365]:
```suggestion
Generate the next message in a chat with a provided model. This is a streaming endpoint, so there will be a series of responses. Streaming can be disabled using `"stream": false`. The final response object will include statistics and additional data from the request.
```

File: docs/api.md [hunk 29]
```diff
@@ -298,20 +313,46 @@ curl http://localhost:11434/api/generate -d '{
...
+  "prompt_eval_count": 26,
+  "prompt_eval_duration": 107345000,
+  "eval_count": 237,
+  "eval_duration": 4289432000
+}
+```
+
+#### Load a model
+
+If you post a body with just the model, it will load the model into memory, but not do a generation.
```
Comment (jmorganca) [inline docs/api.md:336]:
```suggestion
If no prompt is provided the model into memory.
```

File: docs/api.md [hunk 30]
```diff
@@ -131,21 +137,75 @@ If `stream` is set to `false`, the response will be a single JSON object:
...
+    "color": "warm gray"
+  },
+  "evening": {
+    "color": "orange"
+  }
 }
 ```
 
-#### Request (with images)
+#### Generate request (with images)
```
Comment (jmorganca) [inline docs/api.md:203]:
```suggestion
#### Request (with images)
```

File: docs/api.md [hunk 31]
```diff
@@ -162,20 +222,21 @@ curl http://localhost:11434/api/generate -d '{
...
+  "load_duration": 2559292,
+  "prompt_eval_count": 1,
+  "prompt_eval_duration": 2195557000,
+  "eval_count": 44,
+  "eval_duration": 736432000
 }
 ```
 
-#### Request (Raw Mode)
+#### Generate request (Raw Mode)
```
Comment (jmorganca) [inline docs/api.md:236]:
```suggestion
#### Request (Raw Mode)
```

File: docs/api.md [hunk 32]
```diff
@@ -131,21 +137,75 @@ If `stream` is set to `false`, the response will be a single JSON object:
...
+  "total_duration": 5043500667,
+  "load_duration": 5025959,
+  "prompt_eval_count": 26,
+  "prompt_eval_duration": 325953000,
+  "eval_count": 290,
+  "eval_duration": 4709213000
+}
+```
+
+#### Generate request (JSON mode)
```
Comment (jmorganca) [inline docs/api.md:151]:
```suggestion
#### Request (JSON mode)
```

File: docs/api.md [hunk 33]
```diff
@@ -131,21 +137,75 @@ If `stream` is set to `false`, the response will be a single JSON object:
...
+
+When `format` is set to `json`, the output will always be a well-formed JSON object. You must also include text like 'respond using JSON' or 'output in JSON' in the prompt. If you include a schema in the prompt listing the desired property names and types, most models tend to use it. In cases where results aren't perfect, few-shot prompting has a positive effect.
```
Comment (jmorganca) [inline docs/api.md:153]:
```suggestion
> When `format` is set to `json`, the output will always be a well-formed JSON object. It's important to also instruct the model to respond in JSON.
```

File: docs/api.md [hunk 34]
```diff
@@ -99,20 +103,22 @@ To calculate how fast the response is generated in tokens per second (token/s),
...
+  "load_duration":6338219291,
+  "prompt_eval_count":26,
+  "prompt_eval_duration":130079000,
+  "eval_count":259,
+  "eval_duration":4232710000
 }
 ```
 
-#### Request (No streaming)
+#### Generate request (No streaming)
```
Comment (jmorganca) [inline docs/api.md:117]:
```suggestion
#### Request (No streaming)
```

File: docs/api.md [hunk 35]
```diff
@@ -99,20 +103,22 @@ To calculate how fast the response is generated in tokens per second (token/s),
...
+
+##### Request
```
Comment (jmorganca) [inline docs/api.md:119]: Do we need this layer (h5) of headers? It's getting deep

File: docs/api.md [hunk 36]
```diff
@@ -17,15 +17,17 @@
...
 ### Durations
 
 All durations are returned in nanoseconds.
 
 ### Streaming responses
 
-Certain endpoints stream responses as JSON objects.
+Certain endpoints stream responses as JSON objects and can optional return non-streamed responses.
+
+<hr>
```
Comment (jmorganca) [inline docs/api.md:30]:
```suggestion
```

File: docs/api.md [hunk 37]
```diff
@@ -17,15 +17,17 @@
 
 ### Model names
 
-Model names follow a `model:tag` format. Some examples are `orca-mini:3b-q4_1` and `llama2:70b`. The tag is optional and, if not provided, will default to `latest`. The tag is used to identify a specific version.
+Model names follow a `model:tag` format. Some examples are `orca-mini:3b-q4_1` and `llama2:70b`. The tag is optional and, if not provided, will default to `latest`. The tag is used to identify a specific version. Models uploaded by users will have the format `namespace/model:tag`.
```
Comment (jmorganca) [inline docs/api.md:20]:
```suggestion
Model names follow a `model:tag` format, where `model` can have an optional namespace such as `example/model`. Some examples are `orca-mini:3b-q4_1` and `llama2:70b`. The tag is optional and, if not provided, will default to `latest`. The tag is used to identify a specific version.
```

File: README.md [hunk 38]
```diff
@@ -120,7 +120,7 @@ ollama run mario
 Hello! It's your friend Mario.
 ```
 
-For more examples, see the [examples](examples) directory. For more information on working with a Modelfile, see the [Modelfile](docs/modelfile.md) documentation.
+For more examples, see the [examples](examples) directory. For more information on working with a Modelfile, see the [Modelfile](docs/modelfile.md) documentation. For details about using the CLI, refer to our [CLI Documentation](./docs/cli.md).
```
Comment (jmorganca) [inline README.md:123]:
```suggestion
For more examples, see the [examples](examples) directory. For more information on working with a Modelfile, see the [Modelfile](docs/modelfile.md) documentation.
```

File: docs/README.md [hunk 39]
```diff
@@ -1,6 +1,27 @@
...
 
-- [Modelfile](./modelfile.md)
-- [How to develop Ollama](./development.md)
-- [API](./api.md)
-- [Tutorials](./tutorials.md)
+All information about Ollama beyond the main README can be found in three main areas:
+
+1. The documentation describes all the features and functionality of Ollama.
+2. [Tutorials](./tutorials.md) apply the documentation to tasks.
+3. [Examples](./examples) include working tools and applications that use Ollama in interesting ways.
```
Comment (jmorganca) [inline docs/README.md:7]: This link might need to be `../examples`

File: docs/README.md [hunk 40]
```diff
@@ -1,6 +1,27 @@
...
+
+## Documentation Topics
+
+Ollama is a tool for running AI models on your hardware. Many users will choose to use the Command Line Interface (CLI) to work with Ollama. Learn more about all the commands in the CLI in the **[Main Readme](../README.md)**.
```
Comment (jmorganca) [inline docs/README.md:11]:
```suggestion
To get started, see the project's **[quicktart](../README.md#quickstart)**.
```

File: docs/README.md [hunk 41]
```diff
@@ -1,6 +1,27 @@
...
+
+Use the RESTful API using any language, including Python, JavaScript, Typescript, Go, Rust, and many more. Learn more about using the API in the **[API Documentation](./api.md)**.
```
Comment (jmorganca) [inline docs/README.md:13]:
```suggestion
Ollama has a REST API. For more information see the [API Documentation](./api.md).
```

File: docs/README.md [hunk 42]
```diff
@@ -1,6 +1,27 @@
...
+
+Create new models or modify models already in the library using the Modelfile. Learn more about the Modelfile syntax in the **[Modelfile Documentation](./modelfile.md)**.
```
Comment (jmorganca) [inline docs/README.md:15]:
```suggestion
Create new models or modify models already in the [library](https://ollama.ai/library) using the Modelfile. Learn more about the Modelfile syntax in the **[Modelfile Documentation](./modelfile.md)**.
```

File: docs/README.md [hunk 43]
```diff
@@ -1,6 +1,27 @@
...
+
+Import models using source model weights found on Hugging Face and similar sites by referring to the **[Import Documentation](./import.md)**.
+
+Installing on Linux in most cases is easy using the script on Ollama.ai. To get more detail about the install, including CUDA drivers, see the **[Linux Documentation](./linux.md)**.
```
Comment (jmorganca) [inline docs/README.md:19]:
```suggestion
For customizing installs of Ollama on Linux see the **[Linux Documentation](./linux.md)**.
```

File: docs/README.md [hunk 44]
```diff
@@ -1,6 +1,27 @@
...
+
+Many of our users like the flexibility of using our official Docker Image. Learn more about using Docker with Ollama using the **[Docker Documentation](./docker.md)**.
```
Comment (jmorganca) [inline docs/README.md:21]:
```suggestion
Ollama can be deployed using Docker. Learn more about using Docker with Ollama using the **[Docker Documentation](./docker.md)**.
```

File: docs/troubleshooting.md (hunk 18 above)
Comment (technovangelist) [reply docs/troubleshooting.md:19]: so not just explaining common messages in the logs

Comment (jmorganca) [reply docs/troubleshooting.md:19]: A combo of error message + link to issue for further discussion would be best!

File: docs/api.md (hunk 27 above)
Comment (technovangelist) [reply docs/api.md:386]: I didn't really have anything to add, but the levels are there for consistency with the rest of the doc

File: docs/README.md [hunk 45]
```diff
@@ -1,6 +1,27 @@
...
 
-- [Modelfile](./modelfile.md)
-- [How to develop Ollama](./development.md)
-- [API](./api.md)
-- [Tutorials](./tutorials.md)
+All information about Ollama beyond the main README can be found in three main areas:
+
+1. The documentation describes all the features and functionality of Ollama.
+2. [Tutorials](./tutorials.md) apply the documentation to tasks.
+3. [Examples](../examples) include working tools and applications that use Ollama in interesting ways.
```
Comment (jmorganca) [inline docs/README.md:7]: Let's add these below and remove `## Documentation Topics`

File: docs/faq.md [hunk 46]
```diff
@@ -1,138 +1,90 @@
 # FAQ
 
-## How can I view the logs?
+## How can I upgrade Ollama
```
Comment (jmorganca) [inline docs/faq.md:3]:
```suggestion
## How can I upgrade Ollama?
```

File: docs/troubleshooting.md [hunk 47]
```diff
@@ -0,0 +1,23 @@
...
+journalctl -u ollama
+```
+
+If manually running `ollama serve` in a terminal, the logs will be on that terminal.
+
+Sign in to the Discord at [discord.gg/ollama](https://discord.gg/ollama) for help interpreting the logs.
+
+## Known issues
+
+This is a list of some of the errors or issues that can pop up and their solutions:
```
Comment (jmorganca) [inline docs/troubleshooting.md:21]:
```suggestion
```

File: docs/README.md [hunk 48]
```diff
@@ -1,6 +1,27 @@
...
+
+## Documentation Topics
```
Comment (jmorganca) [inline docs/README.md:9]:
```suggestion
## Topics
```

File: docs/README.md [hunk 49]
```diff
@@ -1,6 +1,27 @@
...
+
```
Comment (jmorganca) [inline docs/README.md:10]:
```suggestion

For working code examples of using Ollama, see [Examples](../examples)

```

File: docs/faq.md [hunk 50]
```diff
@@ -1,138 +1,90 @@
 # FAQ
 
-## How can I view the logs?
+## How can I upgrade Ollama?
 
-On macOS:
+To upgrade Ollama, run the installation process again. On the Mac, click the Ollama icon in the menubar and choose the restart option if an update is available.
```
Comment (jmorganca) [inline docs/faq.md:5]:
```suggestion
To upgrade Ollama, run the installation process again. On the macOS, click the Ollama icon in the menubar and choose the restart option if an update is available.
```

Comment (technovangelist) [reply docs/faq.md:5]: on macOS, not the macOS

File: docs/troubleshooting.md [hunk 51]
```diff
@@ -0,0 +1,22 @@
...
+
+On Linux systems with systemd, the logs can be found with this command:
+
+```shell
+journalctl -u ollama
+```
+
+If manually running `ollama serve` in a terminal, the logs will be on that terminal.
+
+Sign in to the Discord at [discord.gg/ollama](https://discord.gg/ollama) for help interpreting the logs.
```
Comment (jmorganca) [inline docs/troubleshooting.md:17]:
```suggestion
Join the [Discord](https://discord.gg/ollama) for help interpreting the logs.
```

File: docs/faq.md [hunk 52]
```diff
@@ -1,138 +1,90 @@
...
+1. Click the menubar icon for Ollama and choose **Quit Ollama**.
+2. Open a new terminal window and run the following command (this example uses `OLLAMA_HOST` with an IP address of `123.1.1.1`):
 
-Ollama binds to 127.0.0.1 port 11434 by default. Change the bind address with the `OLLAMA_HOST` environment variable.
+   ```bash
+   OLLAMA_HOST=123.1.1.1 ollama serve
+   ```
 
-On macOS:
+## How do I use Ollama server environment variables on Linux
```
Comment (jmorganca) [inline docs/faq.md:22]:
```suggestion
## How do I use Ollama server environment variables on Linux?
```

File: README.md [hunk 53]
```diff
@@ -128,6 +128,10 @@ For more examples, see the [examples](examples) directory. For more information
 
 `ollama create` is used to create a model from a Modelfile.
 
+```
+ollama create updatedllama -f ./Modelfile
```
Comment (jmorganca) [inline README.md:132]:
```suggestion
ollama create mymodel -f ./Modelfile
```

File: docs/troubleshooting.md [hunk 54]
```diff
@@ -0,0 +1,22 @@
...
+```
+
+If manually running `ollama serve` in a terminal, the logs will be on that terminal.
+
+Join the [Discord](https://discord.gg/ollama) for help interpreting the logs.
+
+## Known issues
+
+
+**signal: illegal instruction (core dumped).** - Ollama requires AVX support from the CPU. This was introduced in 2011 and CPUs started offering it in 2012. CPUs from before that and some lower end CPUs after that may not have AVX support and thus are not supported by Ollama. Some users have had luck with building Ollama on their machines disabling the need for AVX.
```
Comment (jmorganca) [inline docs/troubleshooting.md:22]:
```suggestion
* `signal: illegal instruction (core dumped)`: Ollama requires AVX support from the CPU. This was introduced in 2011 and CPUs started offering it in 2012. CPUs from before that and some lower end CPUs after that may not have AVX support and thus are not supported by Ollama. Some users have had luck with building Ollama on their machines disabling the need for AVX.
```

**Review by jmorganca — APPROVED** [review]
